### PR TITLE
S3 glacier documentation + RFC

### PR DESCRIPTION
**Pull request by jvarho**
Hi,

I'm using rclone to back up stuff into an S3 bucket and was playing around with [lifecycle policies](http://docs.aws.amazon.com/AmazonS3/latest/user-guide/create-lifecycle.html) in an attempt to reduce storage costs. It is easy to make old files automatically move to lower cost glacier storage (though I've yet to verify if the increased cost per operation makes it worth it in my case).

When syncing to S3 everything works fine, but when syncing/copying back objects would need to be manually restored before rclone can download them. This PR only adds documentation and an error message to that effect.

I could look into adding actual restore support but I'm not sure how it should work. Since it takes minutes to hours after restore for the files to be available, it cannot really be done as part of sync/copy - it is basically a separate operation from the normal commands. Perhaps it could be a flag to check? Something like `rclone check --S3-restore-nonmatching`... though that does not seem quite right.

### CODE REVIEW

File: docs/content/s3.md [hunk 1]
```diff
@@ -299,6 +299,14 @@ Notes on above:
 For reference, [here's an Ansible script](https://gist.github.com/ebridges/ebfc9042dd7c756cd101cfa807b7ae2b) 
 that will generate one or more buckets that will work with `rclone sync`.
 
+### Glacier ###
+
+You can transition objects to glacier storage using a [lifecycle policy](http://docs.aws.amazon.com/AmazonS3/latest/user-guide/create-lifecycle.html).
+The bucket can still be synced or copied into normally, but if rclone
+tries to access the data you will see an error. In that case you need to
```
Comment (ncw) [inline docs/content/s3.md]: Perhaps mention what the error is here since you put it in the other commit?

**Comment by ncw** [comment]
> I could look into adding actual restore support but I'm not sure how it should work. Since it takes minutes to hours after restore for the files to be available, it cannot really be done as part of sync/copy - it is basically a separate operation from the normal commands. Perhaps it could be a flag to check? Something like rclone check --S3-restore-nonmatching... though that does not seem quite right.

How about something like...  We make a new flag for s3, say `--s3-glacier-restore` which means if an object in the sync needs restoring then start a restore, but return an error.

You would run your sync with this flag and it would error out, however in the process it would ask for all the objects needed to be restored.

It would be up to the user to run the sync again in the future (maybe rclone could say about how long to wait in a log message if known)  when the objects were restored and the sync would then complete properly.

That would be a bit manual, but it would enable working with glacier.

**Comment by jvarho** [comment]
@ncw I addressed your review.

I'll try implementing restore later this week if I find the time. Based on your comments I agree that a flag to copy/sync which restores would be best. I think it should copy/sync what it can, start restores for the rest and report an error at the end.

**Comment by ncw** [comment]
Thanks for that - I've merged it now :-)

Look forward to seeing your PR!

-- Nick

**Pull request merged**